VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -441,23 +446,17 @@ private void hideProgress() {
     public void onSaveInstanceState(Bundle outState) {
         logger.debug("Saving state ...");
         stateSaved = true;
-        if(player!=null){
-            // hold on until activity is being destroyed, otherwise we assume next call would be restart()
-            boolean changingConfig = getActivity().isChangingConfigurations();
-            logger.debug("Player fragment changing config?  =" + changingConfig);
-            if ( !changingConfig) {
-                // you MUST PAUSE the video  
-                // only if screen is stopping due to any reason other than CONFIGURATION CHANGE
-                player.setPausedOnUnfreeze();
-            }
-            
+        if (player != null) {
             freezePlayer();
-            outState.putSerializable("player", player);
+            outState.putSerializable(KEY_PLAYER, player);
         }
+        outState.putBoolean(KEY_PREPARED, isPrepared);
+        outState.putBoolean(KEY_AUTOPLAY_DONE, isAutoPlayDone);
+        //FIXME: ensure that prepare is called on all activity restarts and then this can be removed
```

Comment:
[MA-1820] (https://openedx.atlassian.net/browse/MA-1820) has been opened to address this issue.